cps/oauth_bb.py

```diff
@@ -22,328 +21,183 @@
+                assert current_user.id == result.id
             else:
-                # bind to current user
-                if current_user and current_user.is_authenticated:
-                    oauth.user = current_user
-                    try:
-                        ub.session.add(oauth)
-                        ub.session.commit()
-                    except Exception as e:
-                        log.exception(e)
-                        ub.session.rollback()
-                    return redirect(url_for('web.login'))
-                #if config.config_public_reg:
-                #   return redirect(url_for('web.register'))
-                #else:
-                #    flash(_(u"Public registration is not enabled"), category="error")
-                #    return redirect(url_for(redirect_url))
-        except NoResultFound:
-            return redirect(url_for(redirect_url))
-
-
-    def get_oauth_status():
-        status = []
-        query = ub.session.query(ub.OAuth).filter_by(
-            user_id=current_user.id,
-        )
-        try:
-            oauths = query.all()
-            for oauth in oauths:
-                status.append(int(oauth.provider))
-            return status
-        except NoResultFound:
-            return None
-
-
-    def unlink_oauth(provider):
-        if request.host_url + 'me' != request.referrer:
-            pass
-        query = ub.session.query(ub.OAuth).filter_by(
-            provider=provider,
-            user_id=current_user.id,
-        )
-        try:
-            oauth = query.one()
-            if current_user and current_user.is_authenticated:
-                oauth.user = current_user
-                try:
-                    ub.session.delete(oauth)
-                    ub.session.commit()
-                    logout_oauth_user()
-                    flash(_(u"Unlink to %(oauth)s success.", oauth=oauth_check[provider]), category="success")
-                except Exception as e:
-                    log.exception(e)
-                    ub.session.rollback()
-                    flash(_(u"Unlink to %(oauth)s failed.", oauth=oauth_check[provider]), category="error")
-        except NoResultFound:
-            log.warning("oauth %s for user %d not fount", provider, current_user.id)
-            flash(_(u"Not linked to %(oauth)s.", oauth=oauth_check[provider]), category="error")
-        return redirect(url_for('web.profile'))
+                log.info("authorized %s: logging in %s", blueprint.name, result)
+                login_user(result)
+
+        # Disable Flask-Dance's default behavior for saving the OAuth token
+        return True
+
+
+    # called by flask-dance on receiving an error from the OAuth provider
+    def error_handler(blueprint, error, error_description=None, error_uri=None):
+        provider = services.oauth.providers.get(blueprint.name)
+        assert provider, "error %s: no provider found" % blueprint.name
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/Ozzie-IsaacsOrganization/calibre-web/pullRequest?prid=4846306)